test/agent/cmd/snat-utils/main.go

```diff
@@ -76,30 +76,40 @@ func validateIPTableRules(randomizedSNATValue string, numOfCidrs int) error {
 
 	currChain := "AWS-SNAT-CHAIN-0"
 	lastChain := fmt.Sprintf("AWS-SNAT-CHAIN-%d", numOfCidrs)
-	i := 0
-	for i < numOfCidrs {
-		rules, err := iptables.List("nat", currChain)
-		if err != nil {
-			return err
-		}
-		i = i + 1
-		nextChain := fmt.Sprintf("AWS-SNAT-CHAIN-%d", i)
-		foundNextChain := false
-		for _, rule := range rules {
-			target := fmt.Sprintf("-j %s", nextChain)
-			if strings.Contains(rule, target) {
-				currChain = nextChain
-				foundNextChain = true
-				break
+
+	rules, err := iptables.List("nat", "AWS-SNAT-CHAIN-1")
+	if err != nil {
+		return err
+	}
+	// If AWS-SNAT-CHAIN-1 has more than one rule, we run the older test agent logic
+	if len(rules) > 1 {
+		i := 0
+		for i < numOfCidrs {
+			rules, err := iptables.List("nat", currChain)
+			if err != nil {
+				return err
+			}
+			i = i + 1
+			nextChain := fmt.Sprintf("AWS-SNAT-CHAIN-%d", i)
+			foundNextChain := false
+			for _, rule := range rules {
+				target := fmt.Sprintf("-j %s", nextChain)
+				if strings.Contains(rule, target) {
+					currChain = nextChain
+					foundNextChain = true
+					break
+				}
+			}
+			if !foundNextChain {
+				return fmt.Errorf("failed: AWS-SNAT chain broken for %s", currChain)
 			}
 		}
-		if foundNextChain == false {
-			return fmt.Errorf("failed: AWS-SNAT chain broken for %s", currChain)
-		}
+	} else {
+		lastChain = "AWS-SNAT-CHAIN-0"
```

Comment:
The old logic counted the number of `AWS-SNAT-CHAIN-%d` chains based on the number of CIDRs in the VPC. We should still do that check, but by counting the number of entries in the `AWS-SNAT-CHAIN-0` chain. That check seems like it can just be added below